cli/pkg/kctrl/local/sources/fetch.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package sources
+
+import (
+	"fmt"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	cmdcore "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/core"
+	buildconfigs "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/local/buildconfigs"
+	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
+)
+
+const (
+	VendirSyncDirectory = "upstream"
+	vendirFileName      = "vendir.yml"
+	includeAllFiles     = "*"
+)
+
+const (
+	FetchFromGithubRelease       string = "Github Release"
+	FetchFromGit                 string = "Git Repository"
+	FetchFromHelmRepo            string = "Helm Chart from Helm Repository"
+	FetchChartFromGit            string = "Helm Chart from Git Repository"
+	FetchFromLocalDirectory      string = "Local Directory"
```

Comment:
I actually did create a type `FetchMode`, with functions `IsLocal()` and `String()`
Since they are directly used as string options while prompting the user directly, it was making things a bit verbose

But I am happy to make the change

Reply:
Removed the prefix for now, leaving this open if we wanna have a type

Reply:
Another case for having a type is that the check `fetchMode == LocalDirectory` is in a bunch of places and we will need to add cases if we introduce more local workflows.
Having something like `IsLocal` which has this condition really helps out , hm.